File: cis_interface/examples/maxMsg/src/maxMsgSrv.m


disp('maxMsgSrv(M): Hello!');
rpc = PsiInterface('PsiRpcServer', 'maxMsgSrv', '%s', '%s');

while (1)
  input = rpc.rpcRecv();
  if (~input{1})
    break;
  end
  fprintf('maxMsgSrv(M): rpcRecv returned %d, input %s\n', ...
	  input{1}, char(input{2}{1}));
  rpc.rpcSend(input{2}{1});
end

disp('maxMsgSrv(M): Goodbye!');

File: cis_interface/examples/maxMsg/src/maxMsgSrv.py
from cis_interface.interface.CisInterface import CisRpcServer


print("maxMsgSrv(P): Hello!")
rpc = CisRpcServer("maxMsgSrv", "%s", "%s")

while True:
    ret, input = rpc.rpcRecv()
    if not ret:
        break
    print("maxMsgSrv(P): rpcRecv returned %s, input %.10s..." % (ret, input[0]))
    flag = rpc.rpcSend(input[0])
    if not flag:
        raise RuntimeError('maxMsgSrv(P): Error sending reply.')

print("maxMsgSrv(P): Goodbye!")
